src/Illuminate/Http/Client/Pool.php

```diff
@@ -33,18 +26,18 @@ class Pool
     /**
      * Create a new requests pool.
      *
-     * @param  \Illuminate\Http\Client\Factory|null  $factory
+     * @param  \Illuminate\Http\Client\PendingRequest|null  $factory
      * @return void
      */
-    public function __construct(Factory $factory = null)
+    public function __construct(PendingRequest $factory = null)
```

Comment:
This is a major breaking change.

Reply:
Only if people are directly creating pools without going through the `Http::pool()` method. Tbh, we could also just remove the type in the constructor and it would work the same. In the end, `$this->factory = ($factory ?? new Factory())->async()->setHandler($handler);` will always be a `PendingRequest`, whether `$factory` is a `Factory` or `PendingRequest`.

If you want, I can split this PR into 2 PR:
1. Resolving the issue with the `setHandler` method on the `PendingRequest` (which prevents middleware from being registered inside pools)
2. Pool setup not being passed to the requests created from the pool

Reply:
Just one question, this change of yours won't break the multiCurlHandler functionality where all pending requests are executed concurrently, right?